### PR TITLE
If data is of uint8, then set plot_settings and norm accordingly

### PR DESCRIPTION
**Pull request by enthusiastic2003**
#7259 If the datatype of the map is uint8, then don't bother creating norm, and create plot_settings accordingly, without the 'norm' key. If data is of any other datatype, then create norm and create plot_settings with the 'norm' key.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,7 +5,6 @@ __pycache__/
 *.py[cod]
 *$py.class
 tmp/
-
```
Comment (nabobalis) [inline .gitignore:8]: This hchange should be reverted.

File: sunpy/map/mapbase.py [hunk 2]
```diff
@@ -235,19 +235,22 @@ def __init__(self, data, header, plot_settings=None, **kwargs):
...
+            self.plot_settings = {'cmap': 'gray',
+                                'norm': norm,
+                                'interpolation': 'nearest',
+                                'origin': 'lower'
+                                }
+        else:
+            self.plot_settings = {'cmap': 'gray',
+                                'interpolation': 'nearest',
+                                'origin': 'lower'
+                                }
```
Comment (nabobalis) [inline sunpy/map/mapbase.py:253]: Is this not repeating itself?

Comment (nabobalis) [reply sunpy/map/mapbase.py:253]: Just seems very repetitive, can we not improve on that?

File: sunpy/map/mapbase.py [hunk 3]
```diff
@@ -235,19 +235,22 @@ def __init__(self, data, header, plot_settings=None, **kwargs):
         # TODO: This should be a function of the header, not of the map
         self._validate_meta()
 
-        if self.dtype == np.uint8:
-            norm = None
-        else:
+        if self.dtype != np.uint8:
```
Comment (ayshih) [inline sunpy/map/mapbase.py:238]:
```suggestion
        self.plot_settings = {'cmap': 'gray',
                            'interpolation': 'nearest',
                            'origin': 'lower'
                            }
        if self.dtype != np.uint8:
```

File: sunpy/map/mapbase.py [hunk 4]
```diff
@@ -235,19 +235,22 @@ def __init__(self, data, header, plot_settings=None, **kwargs):
...
             # Put import here to reduce sunpy.map import time
             from matplotlib import colors
             norm = colors.Normalize()
 
         # Visualization attributes
```
Comment (ayshih) [inline sunpy/map/mapbase.py:243]:
```suggestion
            self.plot_settings['norm'] = colors.Normalize()
```

File: sunpy/map/mapbase.py (hunk 2 above)
Comment (ayshih) [inline sunpy/map/mapbase.py:253]:
```suggestion
```

**Review by ayshih — COMMENTED** [review]
This is one way to detangle the logic and remove the code duplication

**Comment by ayshih** [comment]
As with #7371, I'd like to see developers first sharing opinions on #7259 about whether we should be continuing to treat uint8 data differently before merging this PR

**Comment by enthusiastic2003** [comment]
@nabobalis , Should i add a unit test case?

**Comment by nabobalis** [comment]
Yes please.

**Review by nabobalis — APPROVED** [review]

**Pull request merged**